app/controllers/sessions_controller.rb

```diff
@@ -1,17 +1,20 @@
 class SessionsController < ApplicationController
-
   before_action :user, only: [:create]
 
   def new
     if current_user
-      redirect_to user_path(current_user), alert: t(".flash.alert")
+      redirect_to users_path, alert: t(".flash.alert")
     end
   end
 
   def create
     if user&.authenticate(params[:session][:password])
       session[:user_id] = user.id
-      redirect_to user_path(user), notice: t(".flash.notice")
+      if user.admin?
+        redirect_to admin_users_path, notice: t(".flash.notice")
+      else
+        redirect_to users_path, notice: t(".flash.notice")
+      end
```

Comment:
Every time I look at this `if`, I have an urge to suggest pulling it out into another method. And then I second guess this decision, because why is it needed if it is only used in one place.
I am still leaning to pull it out into a method that describe what it does, but really that's up to you

If I was to do that, I would take note from how devise does that, for example:
```
def after_sign_in_path_for(user)
  if user.admin?
    admin_users_path
  else
    users_path
  end
end
```

```
# controller:
redirect_to after_sign_in_path_for(user), notice: t(".flash.notice")
```

Reply:
Very cool, wouldn't have thought you could do this - done in sha: db0a541